pkg/loadbalancer/experimental/writer.go

```diff
@@ -469,14 +480,18 @@ func (w *Writer) updateBackends(txn WriteTxn, serviceName loadbalancer.ServiceNa
 }
 
 func (w *Writer) DeleteBackendsOfService(txn WriteTxn, name loadbalancer.ServiceName, src source.Source) error {
+	return w.DeleteBackendsOfServiceFromCluster(txn, name, src, LocalClusterID)
+}
+
+func (w *Writer) DeleteBackendsOfServiceFromCluster(txn WriteTxn, name loadbalancer.ServiceName, src source.Source, clusterID uint32) error {
 	for be := range w.bes.List(txn, BackendByServiceName(name)) {
 		if inst := be.GetInstanceFromSource(name, src); inst != nil {
-			be, orphaned := be.releasePerSource(name, src)
+			beNew, orphaned := be.releasePerSource(name, src, clusterID)
 			var err error
 			if orphaned {
-				_, _, err = w.bes.Delete(txn, be)
-			} else {
-				_, _, err = w.bes.Insert(txn, be)
+				_, _, err = w.bes.Delete(txn, beNew)
+			} else if beNew != be {
```

Comment:
Nit. This optimization could be moved to a separate commit.

Reply:
Not an optimization. The logic changed in `releasePerSource` due to skipping backends that have different clusterID leading to it return `be` unchanged (which was not the case here before for this method).